src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -210,3 +214,35 @@ object RetrieveByExternalIdsWithIgnoreUnknownIds {
       value => value.items
     )
 }
+
+trait RetrieveByInstanceIds[R, F[_]] extends WithRequestSession[F] with BaseUrl {
+  def retrieveByInstanceIds(instanceIds: Seq[CogniteInstanceId]): F[Seq[R]]
+  def retrieveByInstanceId(instanceId: CogniteInstanceId): F[R] =
+    retrieveByInstanceIds(Seq(instanceId))
+      .flatMap(items =>
+        F.fromOption(
+          items.headOption,
+          SdkException("Unexpected empty response when retrieving item by InstanceId")
+        )
+      )
+}
+
+trait RetrieveByInstanceIdsWithIgnoreUnknownIds[R, F[_]] extends RetrieveByInstanceIds[R, F] {
+  override def retrieveByInstanceIds(ids: Seq[CogniteInstanceId]): F[Seq[R]] =
+    retrieveByInstanceIds(ids, ignoreUnknownIds = false)
```

Comment:
nit: interesting that ignoreUnknownIds is not lifted to be a parameter, but same in old code too